src/System.Linq/tests/ToArrayTests.cs

```diff
@@ -199,5 +199,95 @@ public void ToArray_ListWhereSelect(int[] sourceIntegers, string[] convertedStri
             Assert.Equal(Array.Empty<string>(), sourceList.Select(i => i.ToString()).Where(s => s == null).ToArray());
         }
 
+        [Fact]
+        public void SameResultsRepeatCallsFromWhereOnIntQuery()
+        {
+            var q = from x in new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 }
+                    where x > Int32.MinValue
+                    select x;
+
+            Assert.Equal(q.ToArray(), q.ToArray());
```

Comment:
I'll add a new test for that specifically, and then a further for empty arrays, as returning the same object in that case is a possible optimisation, so it could come up in someone trying to improve the code, but has potential risks, so it would have to be discussed first.